### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by profi248**
implements:
- encrypted & authenticated MQTT messages
- registration of node to coordinator
- storage of credentials on node and recovery from passphrase

### CODE REVIEW

File: utils/src/bits.rs [hunk 1]
```diff
@@ -156,8 +156,6 @@ fn test_iter_n_bits() {
     let vec: Vec<u16> = vec![0b00000_101_0111_0011, 0b00000_011_0101_1000];
     let mut iter = vec.iter_n_bits(11);
 
-    let mut counter = 0;
```
Comment (nebneb0703) [inline utils/src/bits.rs]: lol this needs to be here

**Review by nebneb0703 — APPROVED** [review]
clion moment

File: utils/src/bits.rs (hunk 1 above)
Comment (profi248) [reply utils/src/bits.rs]: thanks CLion

**Comment by nebneb0703** [comment]
ignoring windows build for now because pain

**Pull request merged**